core/common/src/main/java/alluxio/ZookeeperURI.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import alluxio.util.URIUtils;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * {@link ZookeeperURI} supports Alluxio on Zookeeper URI.
+ */
+@ThreadSafe
+public final class ZookeeperURI extends StandardURI {
```

Comment:
It seems like overkill to create a whole new type of URI just to handle the Zookeeper authority type. Class extension adds a lot of complexity, and shouldn't be used unless there is significant benefit in terms of abstraction and code re-use. I think it would be cleaner to handle the Zookeeper case as a new type of Authority instead of a new type of URI. That would have the added benefit of allowing us to support the Zookeeper authority type for `MultiPartSchemeURI`.

Reply:
Make sense, I should minimize the codes that need to extend.

Change to interface Authority, have the factory create, and getWholeAuthority() and getConnectionAddress() methods.

HostnameAuthority extends Authority, ZookeeperAuthority extends HostnameAuthority.

Thanks!